### DISCUSSION TITLE
Flutter - Publishing failed :| Failed to publish emerg_app.ipa to App Store Connect. Google Play failed to upload artefacts. Package not found:

### DISCUSSION THREAD

**Discussion by S-A24** [codemagic-ci-cd/codemagic-docs#1677] [Q&A] [answered]
### I'm having problems building my Flutter app for it's first release after changing the bundle ID in the following locations:

android/app/build.gradle,  android/app/google-services.json,android/app/src/debug/AndroidManifest.xml, android/app/src/main/AndroidManifest.xml, android/app/src/main/kotlin/com/example/emerg_app/MainActivity.kt, android/app/src/profile/AndroidManifest.xml.

I've cleaned the project, cache, rebuilt the app, pushed changes and double-checked that there's no traces of the old bundle id in the project. The build ID is 6465223af4755e8d4590fbec and the following Artifacts are available:

app-release.aab
mapping.txt
emerg_app.ipa
Runner.app.zip
Runner.app.dSYM.zip

**I'm getting the following error:**
Publishing failed :| Failed to publish emerg_app.ipa to App Store Connect. Google Play failed to upload artefacts. Package not found: com.example.emergApp.: { "error": { "code": 404, "message": "Package not found: com.example.emergApp.", "status": "NOT_FOUND" } }

**Log:**
Failed to upload archive at "/Users/builder/clone/build/ios/ipa/emerg_app.ipa"
Failed to publish /Users/builder/clone/build/ios/ipa/emerg_app.ipa

Failed to publish emerg_app.ipa to App Store Connect.
Publishing app-release.aab to Google Play
Google Play failed to upload artefacts. Package not found: com.example.emergApp.: {
    "error": {
        "code": 404,
        "message": "Package not found: com.example.emergApp.",
        "status": "NOT_FOUND"
    }
}

Build failed :|

Publishing failed :|
Failed to publish emerg_app.ipa to App Store Connect.
Google Play failed to upload artefacts. Package not found: com.example.emergApp.: {
    "error": {
        "code": 404,
        "message": "Package not found: com.example.emergApp.",
        "status": "NOT_FOUND"
    }
}

The reason why I changed the bundle ID's was that in a previous build I got the error below, so I decided to change all the bundle ID's to match to prevent build errors.

Publishing failed :| Failed to publish emerg_app.ipa to App Store Connect. Google Play failed to upload artefacts. Package not found: com.emerg_app.: { "error": { "code": 404, "message": "Package not found: com.emerg_app.", "status": "NOT_FOUND" } }

**Thanks in advance for your help!**

**Comment by dtrdic** [comment]
Hey @sharon-atim, let's first go through google play publishing issue.

When you change the bundle ID (package name) of your app, Google Play Store treats it as a new app rather than an update to an existing app. This means you cannot directly publish the updated version with the new bundle ID as an update to the existing app on Google Play.
Instead, you will need to create a new app listing on Google Play for the updated version with the new bundle ID. This new app listing will be separate from the existing app on Google Play.
So, 
1. create a new app on google play
2. manually upload the created .aab artifact with this new bundle ID to that app

let's go through this first

**Reply by S-A24** [reply]
Hi @dtrdic , thanks for responding. So I created a new app on Google play and I tried adding the created app-release.aab to my google play production release and got this error "You uploaded an APK or Android App Bundle that was signed in debug mode. You need to sign your APK or Android App Bundle in release mode. [Find out more about signing](https://developer.android.com/tools/publishing/app-signing.html).".

I also tried updating the build.gradle to include: 

signingConfigs {
    release {
        keyAlias 'upload'
        keyPassword System.getenv('KEYSTORE_PASSWORD')
        storeFile file('keystore/upload-keystore.jks')
        storePassword System.getenv('KEYSTORE_PASSWORD')
    }
}

Which resulted in errors to do with the storeFile file path and changed it to app/keystore/upload-keystore.jks, same error so I just removed the code and reverted it back. 

Many thanks for your help with this

**Reply by dtrdic** [reply, marked as answer]
hey Sharon, yes you need to sign your bundle in release mode. In your app/build.gradle file, there is buildTypes block, you can define different build types for your app, such as "debug" and "release." Each build type can have its own configurations, and in your case you need to change signingConfigs to release

```
    buildTypes {
        release {
            signingConfig signingConfigs.release
        }
    }
```

**Reply by S-A24** [reply]
Hi @dtrdic , thanks for your help. I managed to fix that issue by adding the following lines:

signingConfigs {
        release {
            keyAlias keystoreProperties['keyAlias']
            keyPassword keystoreProperties['keyPassword']
            storeFile keystoreProperties['storeFile'] ? file(keystoreProperties['storeFile']) : null
            storePassword keystoreProperties['storePassword']
        }
    }
    buildTypes {
        release {
            signingConfig signingConfigs.release
        }
    }

In addition to this, I added the key.properties file and upload-keystore.jks file to my Flutter project and managed to eventually release a buld for android and ios separately.
